Project2015To2017/ProjectConverter.cs

```diff
@@ -52,25 +59,21 @@ public IEnumerable<Project> Convert(string target)
 			var extension = Path.GetExtension(target) ?? throw new ArgumentNullException(nameof(target));
 			if (extension.Length > 0)
 			{
-				extension = extension.ToLowerInvariant();
-				switch (extension)
+				if (extension == ".sln")
 				{
-					case ".sln":
-						foreach (var project in ConvertSolution(target))
-						{
-							yield return project;
-						}
-
-						break;
-
-					case ".csproj":
-						var file = new FileInfo(target);
-						yield return this.ProcessFile(file, null);
-						break;
-
-					default:
+					foreach (var project in ConvertSolution(target))
+					{
+						yield return project;
+					}
+				}
+				else if (ProjectFileMappings.TryGetValue(extension, out var fileExtension))
```

Comment:
`switch` (with `when`) would still seem nicer, but that's not a real issue to be fixed.